xcdat/regridder/regrid2.py

```diff
@@ -46,259 +46,170 @@ def __init__(self, input_grid: xr.Dataset, output_grid: xr.Dataset, **options: A
         """
         super().__init__(input_grid, output_grid, **options)
 
-        self._src_lat = self._input_grid.bounds.get_bounds("Y")
-        self._src_lon = self._input_grid.bounds.get_bounds("X")
-
-        self._dst_lat = self._output_grid.bounds.get_bounds("Y")
-        self._dst_lon = self._output_grid.bounds.get_bounds("X")
-
-        self._lat_mapping: Any = None
-        self._lon_mapping: Any = None
-
-        self._lat_weights: Any = None
-        self._lon_weights: Any = None
-
     def vertical(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
         """Placeholder for base class."""
         raise NotImplementedError()
 
     def horizontal(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
         """See documentation in :py:func:`xcdat.regridder.regrid2.Regrid2Regridder`"""
-        input_data_var = ds.get(data_var, None)
-
-        if input_data_var is None:
+        try:
+            input_data_var = ds[data_var]
+        except KeyError:
             raise KeyError(
-                f"The data variable '{data_var}' does not exist in the dataset."
-            )
+                f"The data variable {data_var!r} does not exist in the dataset."
+            ) from None
 
-        # Do initial mapping between src/dst latitude and longitude.
-        if self._lat_mapping is None and self._lat_weights is None:
-            self._lat_mapping, self._lat_weights = _map_latitude(
-                self._src_lat, self._dst_lat
-            )
+        src_lat_bnds = _get_bounds_ensure_dtype(self._input_grid, "Y")
+        src_lon_bnds = _get_bounds_ensure_dtype(self._input_grid, "X")
 
-        if self._lon_mapping is None and self._lon_weights is None:
-            self._lon_mapping, self._lon_weights = _map_longitude(
-                self._src_lon, self._dst_lon
-            )
+        dst_lat_bnds = _get_bounds_ensure_dtype(self._output_grid, "Y")
+        dst_lon_bnds = _get_bounds_ensure_dtype(self._output_grid, "X")
 
         src_mask = self._input_grid.get("mask", None)
 
         # apply source mask to input data
         if src_mask is not None:
-            input_data_var = input_data_var.where(src_mask != 0.0)
-
-        # operate on pure numpy
-        input_data = input_data_var.values
+            masked_value = self._input_grid.attrs.get("_FillValue", None)
 
-        axis_variable_name_map = {x: y[0] for x, y in input_data_var.cf.axes.items()}
+            if masked_value is None:
+                masked_value = self._input_grid.attrs.get("missing_value", 0.0)
 
-        output_axis_sizes = self._output_axis_sizes(input_data_var)
+            # Xarray defaults to masking with np.nan, CDAT masked with _FillValue or missing_value which defaults to 1e20
+            input_data_var = input_data_var.where(src_mask != 0.0, masked_value)
```

Comment:
Are these lines of code the same discussion we had about using `_FillValue` to represent missing values like in CDAT's regrid2?

Reply:
No handling missing data in the input will be addressed in another PR, this section is specific to masking values in the input data.